Expand install instructions (#524)
developer-portal/content: tech/languages/rust/rust-installation.md

# Rust installation

> ## Prepackaged Rust

Reviewer: ```suggestion
## Rust
```
IMO either just Rust or Packaged Rust or maybe Fedora Distributed Rust? Prepackaged sounds a bit weird to me.

> Fedora's repositories offer prepackaged versions of Rust and its package manager `cargo`, which can be installed as follows:

Reviewer: ```suggestion
Fedora's repositories offer packaged versions of Rust and its package manager, which can be installed as follows:
```
Mention cargo by name later with rest of the tooling that gets pulled in.

```
$ sudo dnf install rust cargo
```

> This command will install the `rustc` compiler, the standard library, gdb support, the `rustdoc` documentation generator and the package manager.

Reviewer: ```suggestion
This command will install the `rustc` compiler, the standard library, gdb support, the `rustdoc` documentation generator and the `cargo` package manager.
```

> ## Via Rust's own installer

Reviewer: ```suggestion
## Installing Rust with rustup
```

> Rust's [official installation instructions](https://www.rust-lang.org/learn/get-started) recommend its own installation mechanism, `rustup`. Rust's official installer is available via `dnf install rustup`. You can run `rustup-init` after installing the package to start the Rust installation.

Reviewer: ````suggestion
Rust's [official installation instructions](https://www.rust-lang.org/learn/get-started) recommend its own installation mechanism, `rustup`.

To install the `rustup` installer run:
```console
$ sudo dnf install rustup
```

Run `rustup-init` after installing the package to start the Rust installation.
````

Use more assertive tone, also more careful usage of the word "Official" if we are installing it from Fedora repos via dnf.

Ideally some default step-by-step instructions for the rustup-init setup would be great, especially if some special care has to be applied somewhere to have it well integrated with Fedora. Also one question that came up as I was reviewing this, can I mix Rust as packaged in Fedora with Rust that I install via rustup?

## Which one do I need?

It depends on your preferences. Some people may prefer prepackaged Rust because it automatically updates via `dnf`. Others may prefer a standardized installation. If you go for prepackaged Rust, keep this in mind:

- `cargo clippy` is not included in Fedora's `cargo` package, but is available via the `clippy` package.
- To enable autocompletions and other code intelligence features in your IDE, you may need to install the `rust-src` and `rustfmt` packages.

> 
> In the next section project creation is discussed.

Reviewer: Hmm, this reads a bit weird, especially when I ask myself where is this next section exactly. I think we can just leave the page without it.

```suggestion
```
